### ISSUE TITLE
[Optimization] Lack of directional lighting pattern / The source of 90% of shadow-casting lights

### ISSUE THREAD

**Issue by ubertpendragon** [Regalis11/Barotrauma#9808]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

A feature request regarding a pretty huge issue when it comes to the amount of shadow-casting lights in submarines:

These are all the available lighting textures from lamps:
![image](https://user-images.githubusercontent.com/73229309/187901273-2c5b5649-2a61-45dd-a677-5890f82c8582.png)

Those textures are either a perfect circle, either a cone with roughly a 40° angle.
Most lamps use a circular lighting pattern, shining light in all directions, which forces builders to use shadow-casting as not to bleed light through walls.

Lamps are often placed on the ceiling with shadow casting light.
Here is what you could achieve with a 180° directional angle
![image](https://user-images.githubusercontent.com/73229309/187901121-97043c3b-56be-45af-8c1d-352d6f504ed0.png)

If you wanted to light an entire room without using shadow-casting, it'd be much more efficient:
![image](https://user-images.githubusercontent.com/73229309/187902838-99f78f28-7902-46ae-843b-92bb2c7eb452.png)

**By using directional light textures, you can significantly reduce the amount of shadow-casting lights inside subs**
How? Introducing new directional lights items, or the option to change the light texture use in Lightcomponent settings.

### Bug prevalence

Just once

### Version

0.18.15.1 / 0.18.15.2 on macOS

### Which operating system did you encounter this bug on?

Windows

**Comment by MaaiG** [comment]
I support! The problem of light sources is quite pressing (1 - https://github.com/Regalis11/Barotrauma/issues/9435, 2 - https://github.com/Regalis11/Barotrauma/discussions/9308, https://github.com/Regalis11/Barotrauma/issues/9364 ). But this proposal does not cover all the options for placement of lamps. For esample - The lamp can be placed in the corner of the room. 
It may be so: 
![187902838-99f78f28-7902-46ae-843b-92bb2c7eb452](https://user-images.githubusercontent.com/93348499/187918114-49c10151-9e5b-454c-9078-2877075cc693.png)

Of course, it would be better if the [sub builder could set the light angle of the lamp by himself](https://github.com/Regalis11/Barotrauma/discussions/8052).

In any case, this is a real another way to solve the problem because my enthusiasm has been greatly diminished due to all these limitations imposed by the game engine.
Finally, the section "Type of lighting" can be added to the lamp settings with a selection of parameters:
- 180 - full circle
- 90 - up
- 90 - down
- 90 to the right
- 90 to the left
- 45 - down and right (as in attached image)
- 45 down and left
- 45 - up and right
- 45 - up and left

The main thing is that all this appeared and developed faster!

**Comment by Ek-Videogames** [comment]
Yeah I explored something like this but came to more or less the same conclusion that it was impossible to reasonably account for the number of different light dimensions needed even on a relatively tame submarine compartmentalization.

**Comment by Ek-Videogames** [comment]
Yeah I explored something like this when the update happened that made lights laggier but came to more or less the same conclusion that it was impossible to reasonably account for the number of different light dimensions needed even on a relatively tame submarine compartmentalization.
